pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -429,15 +450,15 @@ func getTPMapMatchingIncomingAffinityAntiAffinity(pod *v1.Pod, nodeInfoMap map[s
 
 	var lock sync.Mutex
 	var firstError error
-	topologyPairsAffinityPodsMaps = newTopologyPairsMaps()
-	topologyPairsAntiAffinityPodsMaps = newTopologyPairsMaps()
-	appendResult := func(nodeName string, nodeTopologyPairsAffinityPodsMaps, nodeTopologyPairsAntiAffinityPodsMaps *topologyPairsMaps) {
+	topologyPairsAffinityPodsMaps = make(topologyToMatchedTermCount)
+	topologyPairsAntiAffinityPodsMaps = make(topologyToMatchedTermCount)
+	appendResult := func(nodeName string, nodeTopologyPairsAffinityPodsMaps, nodeTopologyPairsAntiAffinityPodsMaps topologyToMatchedTermCount) {
 		lock.Lock()
```

Comment:
lock unlock can move into the IF blocks. or find a way to avoid getting into deep into the function ONLY when needed